### PR TITLE
[CausalLM] Clean up CachedSlim MoE layer: remove unused expert mask and skip redundant decode input allocation

### PR DESCRIPTION
**Pull request by Jungwon-Lee**
## Dependency of the PR

None.

## Commits to be reviewed in this PR

<details><summary>e35e5196 — [CausalLM] Remove unused CachedSlim expert mask</summary><br />

[CausalLM] Remove unused CachedSlim expert mask

Stop requesting and tracking the expert-mask scratch tensor because CachedSlim dispatch builds assignments directly from router indices and never reads the mask.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jungwon-Lee <jungone.lee@samsung.com>

</details>

<details><summary>44bdb13c — [CausalLM] Skip CachedSlim decode input allocation</summary><br />

[CausalLM] Skip CachedSlim decode input allocation

Allocate the gathered expert input only for multi-token batches. Single-token decode now uses the shared input view without first allocating and discarding a temporary tensor.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jungwon-Lee <jungone.lee@samsung.com>

</details>

### Summary

- Remove the unused expert-mask scratch tensor from CachedSlim MoE layer, since dispatch builds assignments directly from router indices and never reads the mask.
- Skip the redundant temporary input allocation during single-token decode; use the shared input view directly, allocating only for multi-token batches.

Signed-off-by: Jungwon-Lee <jungone.lee@samsung.com>

### CODE REVIEW

File: Applications/CausalLM/models/qwen3_cached_slim_moe/qwen_moe_layer_cached.cpp [hunk 1]
```diff
@@ -160,12 +159,6 @@ void CachedSlimMoELayer::finalize(nntrainer::InitLayerContext &context) {
     context.requestTensor({total_tokens, 1, 1, num_experts}, "router_logits",
                           nntrainer::Initializer::NONE, false,
                           nntrainer::TensorLifespan::FORWARD_FUNC_LIFESPAN);
-
-  // Expert mask: [num_experts, batch*seq]
-  expert_mask_idx =
-    context.requestTensor({num_experts, 1, topk, total_tokens}, "expert_mask",
-                          nntrainer::Initializer::ZEROS, false,
-                          nntrainer::TensorLifespan::FORWARD_FUNC_LIFESPAN);
```
Comment (Jungwon-Lee) [inline Applications/CausalLM/models/qwen3_cached_slim_moe/qwen_moe_layer_cached.cpp:168]: `expert_mask_idx ` is never used.

**Review by jaemini-shin — COMMENTED** [review]
Looks like same approach can be applied to gpt_oss_moe_layer_cache, too. Isn't this one included in scope?

File: Applications/CausalLM/models/qwen3_cached_slim_moe/qwen_moe_layer_cached.cpp [hunk 2]
```diff
@@ -198,15 +191,16 @@ inline void CachedSlimMoELayer::compute_expert_forward(
   nntrainer::Tensor gate_out(intermediate_dim);
   nntrainer::Tensor acti_out(intermediate_dim);
   nntrainer::Tensor up_out(intermediate_dim);
-  nntrainer::Tensor token_input(token_input_dim);
+  nntrainer::Tensor token_input;
   // Down projection using optimized dot operation
   nntrainer::Tensor token_expert_output(token_output_dim);
-
+
```
Comment (jaemini-shin) [inline Applications/CausalLM/models/qwen3_cached_slim_moe/qwen_moe_layer_cached.cpp]: This white space should not have passed the cpp-linter...

Comment (jaemini-shin) [reply Applications/CausalLM/models/qwen3_cached_slim_moe/qwen_moe_layer_cached.cpp]: #4160 can handle this issue

Comment (Jungwon-Lee) [reply Applications/CausalLM/models/qwen3_cached_slim_moe/qwen_moe_layer_cached.cpp]: I fixed it. thank you

**Comment by Jungwon-Lee** [comment]
> Looks like same approach can be applied to gpt_oss_moe_layer_cache, too. Isn't this one included in scope?

Thank you for your comment. 
Now, these modifications are also applied to gpt_oss.

**Review by jayden0701 — APPROVED** [review]
LGTM!

**Review by baek2sm — APPROVED** [review]
LGTM

**Review by jaemini-shin — APPROVED** [review]
LGTM!
